src/main/java/com/sap/sgs/phosphor/fosstars/tool/PrettyPrinter.java

```diff
@@ -16,24 +29,84 @@
+    FEATURE_CLASS_TO_NAME.put(ProjectPopularityScore.class, "Project popularity");
+    FEATURE_CLASS_TO_NAME.put(ProjectSecurityAwarenessScore.class, "Security awareness");
+    FEATURE_CLASS_TO_NAME.put(ProjectSecurityTestingScore.class, "Security testing");
+    FEATURE_CLASS_TO_NAME.put(UnpatchedVulnerabilitiesScore.class, "Unpatched vulnerabilities");
+    FEATURE_CLASS_TO_NAME.put(VulnerabilityLifetimeScore.class, "Vulnerability lifetime");
+  }
+
+  /**
+   * Maps a feature to its shorter name which should be used in output.
+   */
+  private static final Map<Feature, String> FEATURE_TO_NAME = new HashMap<>();
+
+  static {
+    FEATURE_TO_NAME.put(OssFeatures.HAS_SECURITY_TEAM, "Does it have a security team?");
+    FEATURE_TO_NAME.put(OssFeatures.HAS_SECURITY_POLICY, "Does it have a security policy?");
+    FEATURE_TO_NAME.put(OssFeatures.VULNERABILITIES, "Info about vulnerabilities");
+    FEATURE_TO_NAME.put(OssFeatures.SECURITY_REVIEWS_DONE, "Security reviews");
+    FEATURE_TO_NAME.put(OssFeatures.IS_APACHE, "Does it belong to Apache?");
+    FEATURE_TO_NAME.put(OssFeatures.IS_ECLIPSE, "Does it belong to Eclipse?");
+    FEATURE_TO_NAME.put(OssFeatures.SUPPORTED_BY_COMPANY, "Is it supported by a company?");
+    FEATURE_TO_NAME.put(
+        OssFeatures.SCANS_FOR_VULNERABLE_DEPENDENCIES, "Does it scan for vulnerable dependencies?");
+  }
+
+  /**
+   * Print out a formatted rating value.
    *
    * @param ratingValue The rating value to be printed.
+   * @return A string to be displayed.
    */
   public String print(RatingValue ratingValue) {
     StringBuilder sb = new StringBuilder();
     sb.append(String.format("[+] Rating: %2.2f out of %2.2f -> %s%n",
         ratingValue.score(), Score.MAX, ratingValue.label()));
     sb.append(String.format("[+] Confidence: %2.2f out of %2.2f%n",
         ratingValue.confidence(), Confidence.MAX));
+    sb.append("[+]\n");
     sb.append(String.format("[+] Here is how the rating was calculated:%n"));
-    sb.append(print(ratingValue.scoreValue(), INDENT_STEP));
+    sb.append(print(ratingValue.scoreValue(), INDENT_STEP, true));
     return sb.toString();
   }
 
-  private String print(ScoreValue scoreValue, String indent) {
+  /**
+   * Print out a formatted score value with a specified indent.
+   *
+   * @param scoreValue The score value to be printed.
+   * @param indent The indent.
+   * @return A string to be displayed.
+   */
+  private String print(ScoreValue scoreValue, String indent, boolean isMainScore) {
     StringBuilder sb = new StringBuilder();
-    sb.append(String.format("[+] %sScore:........%s%n", indent, scoreValue.score().name()));
+
+    if (!isMainScore) {
+      sb.append(String.format("[+] %sSub-score:....%s%n", indent, nameOf(scoreValue.score())));
+    } else {
+      sb.append(String.format("[+] %sScore:........%s%n", indent, nameOf(scoreValue.score())));
+    }
+
+    if (!scoreValue.score().description().isEmpty()) {
```

Comment:
Is there a possiblity of scoreValue or  scoreValue.score() being null here?

Reply:
Fortunately, no. The `scoreValue.score()` can't return null because the constructor of the `ScoreValue` checks the score for null. `scoreValue` can't be null either because it comes from the `RatingValue` which also checks its score value for null in the constructor.